nodes/parachain/src/command.rs

```diff
@@ -259,20 +261,80 @@ pub fn run() -> Result<()> {
 			})
 		}
 		Some(Subcommand::Revert(cmd)) => {
-			construct_async_run!(|components, cli, cmd, config| Ok(cmd.run(components.client, components.backend)))
+			construct_async_run!(|components, cli, cmd, config| Ok(cmd.run(
+				components.client,
+				components.backend,
+				None
+			)))
 		}
 		Some(Subcommand::Benchmark(cmd)) => {
-			if cfg!(feature = "runtime-benchmarks") {
-				let runner = cli.create_runner(cmd)?;
-				match cli.runtime.as_str() {
-					"peregrine" => runner.sync_run(|config| cmd.run::<Block, PeregrineRuntimeExecutor>(config)),
-					"spiritnet" => runner.sync_run(|config| cmd.run::<Block, SpiritRuntimeExecutor>(config)),
-					_ => Err("Unknown runtime".into()),
-				}
-			} else {
-				Err("Benchmarking wasn't enabled when building the node. \
+			let runner = cli.create_runner(cmd)?;
+
+			let runtime = match cli.runtime.as_str() {
+				"peregrine" => Ok(ParachainRuntime::Peregrine),
+				"spiritnet" => Ok(ParachainRuntime::Spiritnet),
+				_ => Err("Unknown runtime"),
+			}?;
+
+			// Switch on the concrete benchmark sub-command
+			match cmd {
```

Comment:
not sure if it's nicer but maybe `match (cmd, cli.runtime.as_str()) {` immediately is easier to read? Not sure tho.

Reply:
Not sure if [65813c4](https://github.com/KILTprotocol/mashnet-node/pull/357/commits/65813c47c89801e169841e58895ac3edbfbc937e) is nicer. Whats your take?

Reply:
I like that the second version doesn't need the small match over the runtime in the beginning. So I think I might like the new version better now. But yeah. It's not a big difference. So as you wish.  Thanks for trying it out tho. 😋